static/src/javascripts/projects/commercial/modules/dfp/create-slots.ts

```diff
@@ -194,61 +193,103 @@ const createAdSlotElements = (
 		adSlot.setAttribute(attr, attrs[attr]);
 	});
 
-	adSlots.push(adSlot);
+	return adSlot;
+};
 
-	return adSlots;
+/**
+ * Split class names and prefix all with ad-slot--${className}
+ */
+const createClasses = (slotName: string, classes?: string): string[] =>
+	[...(classes?.split(' ') ?? []), slotName].map(
+		(className: string) => `ad-slot--${className}`,
+	);
+
+/**
+ * Given default size mappings and additional size mappings from
+ * the createAdSlot options parameter.
+ *
+ * 1. Check that the options size mappings use known device names
+ * 2. If so merge the size mappings
+ *
+ */
+const mergeSizeMappings = (
+	defaultSizeMappings: SizeMappings,
+	optionSizeMappings: CreateSlotOptions['sizes'],
+): SizeMappings => {
+	if (!optionSizeMappings) return defaultSizeMappings;
+	const mergedSizeMappings: SizeMappings = { ...defaultSizeMappings };
+	const optionDevices = Object.keys(optionSizeMappings);
+	for (let i = 0; i < optionDevices.length; i++) {
+		const device = optionDevices[i];
+		const optionSizeMappingsForDevice = optionSizeMappings[device];
+		const defaultSizeMappingsForDevice = defaultSizeMappings[device];
+		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- update tsconfig: "noUncheckedIndexedAccess": true
+		if (defaultSizeMappingsForDevice && optionSizeMappingsForDevice) {
+			mergedSizeMappings[device] = mergedSizeMappings[device].concat(
+				optionSizeMappingsForDevice,
+			);
+		}
+	}
+	return mergedSizeMappings;
 };
 
-export const createSlots = (
+/**
+ * Convert size mappings to a string that will be added to the ad slot
+ * data atttributes.
+ *
+ * e.g. { desktop: [[320,250], [320, 280]] } => { desktop: '320,250|320,280' }
+ *
+ */
+const covertSizeMappingsToStrings = (
+	sizeMappings: SizeMappings,
+): Record<string, string> =>
+	Object.entries(sizeMappings).reduce(
+		(result: Record<string, string>, [device, sizes]) => {
+			result[device] = sizes.join('|');
+			return result;
+		},
+		{},
+	);
+
+/**
+ * Prefix all object keys with data-${key}
+ */
+const createDataAttributes = (
```

Comment:
Similar to `createClasses` in this function you could also narrow the return type to 
```
Record<`data-${string}`, string>`
```
 but I think it's up to personal preference whether to use these types over just strings.